src/engine/x86-64/X86_64MacroAssembler.v3

```diff
@@ -723,6 +723,14 @@ class X86_64MacroAssembler extends MacroAssembler {
 		asm.icall_r(scratch);
 		recordRetSourceLoc();
 	}
+	def emit_call_MemoryReadProbe_fire(probe: MemoryReadProbe) {
+		var codePtr = CiRuntime.unpackClosure<MemoryReadProbe, (u64, u64, u64), void>(probe.fire_probe).0;
+		var refOffset = asm.movq_r_p(Target.V3_PARAM_GPRS[0], Pointer.atObject(probe) - Pointer.NULL);
+		addEmbeddedRefOffset(refOffset);
+		asm.movq_r_l(scratch, codePtr - Pointer.NULL); // XXX: make direct call to runtime if within 2GB
```

Comment:
IIRC think the same XXX appears elsewhere. What about factoring out a helper routine that does it?